source/docs/advanced/performance/cache-control-and-cdn.md

```diff
@@ -0,0 +1,76 @@
+---
+id: cache-control-and-cdn
+title: Cache Control and CDN
+---
+
+<blockquote class="feature--new">
+  _This feature has been added in version `2.6.0`_
+</blockquote>
+
+To enable CDN caching you need to first set up proper [cache headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) for your site.
+
+In Front-Commerce we have implemented a mechanism where you can easily set up the `cache-control` headers for different parts of your site. You can also provide a custom implementation of the `cache-control` if needed (e.g. products with old last modified have longer caching duration).
+
+# Category/Product/CMS Pages
+
+To set up cache control headers for category/product/cms pages, first override `src/web/theme/pages/Category/CategoryCacheControlFragment.gql`, `src/web/theme/pages/Product/ProductCacheControlFragment.gql`, `src/web/theme/pages/CmsPage/CmsPageCacheControlFragment.gql`. Then uncomment the `cacheControl` part of the query. Finally set the `sMaxAge` and `staleWhileRevalidate` values to meet your requirements. So your query should look like this:
+
+```gql
+fragment CategoryCacheControlFragment on Category {
+  cacheControl(input: { sMaxAge: 600, staleWhileRevalidate: 3600 }) {
+    sMaxAge
+    staleWhileRevalidate
+  }
+}
+```
+
+P.S. we use `sMaxAge` instead of `maxAge` because this is the header used by CDN cache.
+
+# Static Pages
+
+To set up cache control headers for static pages you need to set a static attribute on the exported component of that page. Please make sure that this attribute is hoisted all the way to be visible on the exported route component (see `src/web/theme/routes/index.js`). Your static page component should look something like this:
+
+```js
+import Home from "theme/pages/Home";
+
+export default Home;
+
+Home.cacheControlDefinition = {
+  sMaxAge: 10,
+  staleWhileRevalidate: 1000,
+};
+```
+
+# Graph QL queries
+
+To enable caching for Graph QL queries you should make your Graph QL queries use the `GET` method. This can be done by adding the following environment variable to your `.env` file:
+
+```
+FRONT_COMMERCE_WEB_USE_GET_FOR_PERSISTED_QUERIES=true
+```
+
+Once you have enabled `GET` for your Graph QL queries the dispatcher will automatically set the values for the `cache-control` header based on the route (category/product/cms page).
+
+# Setting up the CDN
+
+Finally after you have set up the appropriate cache controls for your site you now need to configure the CDN. Front-Commerce requires one setting on the CDN to work properly and that is to exclude the cache for requests with a cookie `connect.sid`. How this can be achieved depends on your CDN. It is however common to have it as an exclusion regular expression, in that case you have to set it to `^connect.sid` (`^` denoting starts with). If you are using Front-Commerce cloud solution this comes pre configured for you so you do not have to do anything.
```

Comment:
We don't want to go too deep into how to configure CDNs to prevent maintaining documentation for each of them. We could improve this part later with links to the documentatino of the most used ones.

```suggestion
Finally after you have set up the appropriate cache controls for your site you now need to configure a CDN (or a reverse proxy such as Nginx) which will take care of caching pages for users. See [Which CDN providers support stale-while-revalidate?](https://www.ctrl.blog/entry/cdn-rfc5861-support.html) for a list of CDN that supports it. As of 2.6, Front-Commerce requires that the CDN ignores the cache for requests containing a cookie named  `connect.sid`. Please refer to your CDN's documentation for more details. **If you are using Front-Commerce Cloud, you have nothing to do** unless you changed default values. Please [contact us](support@front-commerce.com) if you need support to enable caching.
```